cmd/tsidp/tsidp_test.go

```diff
@@ -824,3 +826,159 @@ func TestExtraUserInfo(t *testing.T) {
 		})
 	}
 }
+
+func TestFunnelClientsPersistence(t *testing.T) {
+	log.SetOutput(io.Discard) // suppress log output during tests
+
+	// Create a temporary file with test funnel clients data
+	testClients := map[string]*funnelClient{
+		"test-client-1": {
+			ID:          "test-client-1",
+			Secret:      "test-secret-1",
+			Name:        "Test Client 1",
+			RedirectURI: "https://example.com/callback",
+		},
+		"test-client-2": {
+			ID:          "test-client-2",
+			Secret:      "test-secret-2",
+			Name:        "Test Client 2",
+			RedirectURI: "https://example2.com/callback",
+		},
+	}
+
+	// Marshal test data to JSON
+	testData, err := json.Marshal(testClients)
+	if err != nil {
+		t.Fatalf("failed to marshal test data: %v", err)
+	}
+
+	// Create a temporary file
+	tmpFile := t.TempDir() + "/oidc-funnel-clients.json"
+	if err := os.WriteFile(tmpFile, testData, 0600); err != nil {
+		t.Fatalf("failed to write test file: %v", err)
+	}
+
+	// Test loading clients when file exists
+	t.Run("loads clients from existing file", func(t *testing.T) {
```

Comment:
try not to use spaces or special chars that need quoting in t.Run names

also, if these aren't independent tests (that can each be run by itself), can you make that clear by naming them like "step1-load-from-file" with "step" in there?